src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -16,10 +17,12 @@ def __init__(
         dependency: Dependency,
         parent: DependencyGraph | None,
         resolver: DependencyResolver,
+        path_lookup: PathLookup,
     ):
         self._dependency = dependency
         self._parent = parent
         self._resolver = resolver
+        self._path_lookup = path_lookup
```

Comment:
```suggestion
        self._path_lookup = path_lookup.change_directory(dependency.path.parent)
```
remove push_cwd/pop_cwd, they're not needed.

Reply:
> the instantiation path is different from the execution path.

I don't believe this is the case. can you address this comment and let's merge this?

Reply:
This is not an opinion, it's an observation, see https://github.com/databrickslabs/ucx/tree/drop-cwd-stack where I did the work you asked for. Tests in test_path_lookup_simulation fail for the reason explained above. 
My wording is inaccurate, it should read: the instantiation sequence is different from the execution sequence.

Reply:
tests are not correct - e.g. you either have to `from .sibling2_file import stuff` (there's a bug in `databricks.labs.ucx.source_code.python_linter.PythonLinter.list_import_sources` that ignores `ast.ImportFrom` level), or add that folder to a syspath.